YoutubeExplode/Playlists/PlaylistClient.cs

```diff
@@ -92,8 +93,20 @@ public async IAsyncEnumerable<Batch<PlaylistVideo>> GetVideoBatchesAsync(
             );
 
             var videos = new List<PlaylistVideo>();
+            IEnumerable<PlaylistVideoData> originalVideos;
+            if (!encounteredIds.Any())
+            {
+                originalVideos = response.Videos.DistinctBy(v => v.Id);
+            }
+            else
+            {
+                originalVideos = response.Videos.Where(v =>
+                {
+                    return !encounteredIds.Any(e => string.Equals(e.Value, v.Id));
+                });
+            }
```

Comment:
I still think the old code and the new code work the same 🤔 

Previously, we used a `HashSet` to deduplicate video IDs. Using the following block:

```csharp
  if (!encounteredIds.Add(videoId))
                    continue; // this will execute if the `videoId` already exists
```

And now we use this instead:

```csharp
            if (!encounteredIds.Any())
            {
                originalVideos = response.Videos.DistinctBy(v => v.Id);
            }
            else
            {
                originalVideos = response.Videos.Where(v =>
                {
                    return !encounteredIds.Any(e => string.Equals(e.Value, v.Id));
                });
            }
```

Which seems to do the same.

Reply:
You can run it by yourself and see that my changes allow to download more videos. I can't explain by words, why this changes works better than code on master tree. My observation just showed that for some reasons, on some point (probably when last item is duplicate), it starts to return videos, that were already added to HashSet.